libraries/core_libs/network/src/tarcap/packets_handlers/pbft_sync_packet_handler.cpp

```diff
@@ -43,9 +66,15 @@ void PbftSyncPacketHandler::process(const PacketData &packet_data, const std::sh
   }
 
   // Process received pbft blocks
-  auto it = packet_data.rlp_.begin();
-  bool last_block = (*it++).toInt<bool>();
-  SyncBlock sync_block(*it);
+  bool last_block = packet_data.rlp_[0].toInt<bool>();
+
+  SyncBlock sync_block;
+  try {
+    sync_block = SyncBlock(packet_data.rlp_[1]);
+  } catch (const Transaction::InvalidSignature &e) {
```

Comment:
? I do not understand the exception here

Reply:
SyncBlock takes as ctor argument rlp bytes and it creates Transaction object from it. Transaction ctor can throw InvalidSignature

Reply:
hmm, I see and any other structures ? like votes/dags/... no-throw?

Reply:
they dont throw in ctors. They rather return error during validation...